src/web/routes/application/are-you-pregnant/test/validate.test.js

```diff
@@ -1,75 +1,113 @@
     }
   }
 
-  t.equal(validateExpectedDeliveryDate('', { req }), true, 'should not validate the date if not pregnant')
+  t.throws(validateExpectedDeliveryDate.bind(null, {}, { req }), /validation:expectedDeliveryDateInvalidTooFarInPast/, 'not in valid date range')
   t.end()
 })
 
-test('addExpectedDeliveryDateToBody() should not add key to body if areYouPregnant is undefined', (t) => {
-  const req = { body: {} }
+test('validateExpectedDeliveryDate() too far in the future', (t) => {
+  const req = {
+    t: (string) => string,
+    body: {
+      areYouPregnant: YES,
+      'expectedDeliveryDate-day': 31,
+      'expectedDeliveryDate-month': 3,
+      'expectedDeliveryDate-year': 9999
+    }
+  }
 
-  addExpectedDeliveryDateToBody(req, res, next)
-  t.equal(typeof req.body.expectedDeliveryDate, 'undefined', 'should not add key to body if areYouPregnant is undefined')
+  t.throws(validateExpectedDeliveryDate.bind(null, {}, { req }), /validation:expectedDeliveryDateInvalidTooFarInFuture/, 'not in valid date range')
   t.end()
 })
 
-test(`addExpectedDeliveryDateToBody() should not add key to body if areYouPregnant is ${NO}`, (t) => {
+test('validateExpectedDeliveryDate() null date', (t) => {
   const req = {
+    t: (string) => string,
     body: {
-      areYouPregnant: NO
+      areYouPregnant: YES,
+      'expectedDeliveryDate-day': null,
+      'expectedDeliveryDate-month': null,
+      'expectedDeliveryDate-year': null
     }
   }
 
-  addExpectedDeliveryDateToBody(req, res, next)
-  t.equal(typeof req.body.expectedDeliveryDate, 'undefined', `should not add key to body if areYouPregnant is ${NO}`)
+  t.throws(validateExpectedDeliveryDate.bind(null, {}, { req }), /validation:expectedDeliveryDateInvalid/, 'not in valid date range')
   t.end()
 })
 
-test(`addExpectedDeliveryDateToBody() should add key to body if areYouPregnant is ${YES}`, (t) => {
+test('validateExpectedDeliveryDate() invalid date', (t) => {
   const req = {
+    t: (string) => string,
     body: {
       areYouPregnant: YES,
-      'expectedDeliveryDate-day': '02',
-      'expectedDeliveryDate-month': '02',
-      'expectedDeliveryDate-year': '2222'
+      'expectedDeliveryDate-day': 'invalid',
+      'expectedDeliveryDate-month': 'invalid',
+      'expectedDeliveryDate-year': 'invalid'
     }
   }
 
-  addExpectedDeliveryDateToBody(req, res, next)
-  t.equal(req.body.expectedDeliveryDate, '2222-02-02', `should add key to body if areYouPregnant is ${YES}`)
+  t.throws(validateExpectedDeliveryDate.bind(null, {}, { req }), /validation:expectedDeliveryDateInvalid/, 'not in valid date range')
   t.end()
 })
+
+test('validateExpectedDeliveryDateNotPregnant()', (t) => {
+  const req = {
+    t: (string) => string,
+    body: {
+      areYouPregnant: NO
+    }
+  }
+
+  t.equal(validateExpectedDeliveryDate({}, { req }), true, 'should not validate the date if not pregnant')
+  t.end()
+})
+
+const createDateWithMonthAdjustment = (monthAdjustment) => {
```

Comment:
Suggestion: I would have this function at the top of the page so there's no issue with it ever not being defined before its called

Reply:
Ok